Lodash All declarations of 'WeakMap' must have identical type parameters.

- [x] I tried using the `@types/lodash` package and had problems.
- [ ] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [ ] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [ ] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @....

I found this problem compiling a project using Typescript version 2.2.0-dev.20170128 and @types/lodash version 4.14.51. In my case the tsconfig use es6 target. The error message is:

`node_modules/@types/lodash/index.d.ts(19421,15): error TS2428: All declarations of 'WeakMap' must have identical type parameters.
`
If I browse the file to the line indicated in the error message I can see the following comment:

`// Backward compatibility with --target es5`

Maybe this is the cause of the issue?
Best regards to all

--- a/lodash/index.d.ts
+++ b/lodash/index.d.ts
@@ -12701,7 +12701,7 @@ declare namespace _ {
          * @param value The value to check.
          * @returns Returns true if value is correctly classified, else false.
          */
-        isWeakMap<K, V>(value?: any): value is WeakMap<K, V>;
+        isWeakMap<K, V>(value?: any): value is WeakMapLike<K, V>;
     }
 
     interface LoDashImplicitWrapperBase<T, TWrapper> {
@@ -19446,5 +19446,5 @@ declare global {
     interface Set<T> { }
     interface Map<K, V> { }
     interface WeakSet<T> { }
-    interface WeakMap<K, V> { }
+    interface WeakMapLike<K, V> { }
 }

--- a/lodash/index.d.ts
+++ b/lodash/index.d.ts
@@ -12701,7 +12701,7 @@ declare namespace _ {
          * @param value The value to check.
          * @returns Returns true if value is correctly classified, else false.
          */
-        isWeakMap<K, V>(value?: any): value is WeakMapLike<K, V>;
+        isWeakMap<K, V>(value?: any): boolean;
     }
 
     interface LoDashImplicitWrapperBase<T, TWrapper> {
@@ -19446,5 +19446,4 @@ declare global {
     interface Set<T> { }
     interface Map<K, V> { }
     interface WeakSet<T> { }
-    interface WeakMapLike<K, V> { }
 }

--- a/lodash/lodash-tests.ts
+++ b/lodash/lodash-tests.ts
@@ -7446,17 +7446,6 @@ namespace TestIsUndefined {
 // _.isWeakMap
 namespace TestIsWeakMap {
     {
-        let value: number|WeakMapLike<string, number>;
-
-        if (_.isWeakMap<string, number>(value)) {
-            let result: WeakMapLike<string, number> = value;
-        }
-        else {
-            let result: number = value;
-        }
-    }
-
-    {
         let result: boolean;
 
         result = _.isWeakMap(any);

--- a/lodash/lodash-tests.ts
+++ b/lodash/lodash-tests.ts
@@ -7446,10 +7446,10 @@ namespace TestIsUndefined {
 // _.isWeakMap
 namespace TestIsWeakMap {
     {
-        let value: number|WeakMap<string, number>;
+        let value: number|WeakMapLike<string, number>;
 
         if (_.isWeakMap<string, number>(value)) {
-            let result: WeakMap<string, number> = value;
+            let result: WeakMapLike<string, number> = value;
         }
         else {
             let result: number = value;

